.changeset/hip-bears-film.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/core': patch
+---
+
+File chunks now keep the filename from provider-generated files (e.g. Gemini) and preserve it when chunks are rebuilt into message parts.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)